src/ripple/rpc/handlers/AccountTx.cpp

```diff
@@ -338,6 +356,11 @@ populateJsonResponse(
                         insertDeliveredAmount(
                             jvObj[jss::meta], context, txn, *txnMeta);
                         insertNFTSyntheticInJson(jvObj, sttx, *txnMeta);
+                        if (auto closeTime =
```

Comment:
Why is the presence of `closeTime` predicated on the existence of `txnMeta` ? Can't we have this block of code at line 341?
```
if (auto closeTime = context.ledgerMaster.getCloseTimeBySeq(txn->getLedger()))
     jvObj[jss::close_time_iso] =
     to_string_iso(*closeTime);
```

Reply:
that's because we only have meta in a validated ledger, and we only show close time in a validated ledger as well

Reply:
I see. It would be more clear if all the fields `jss::ledger_index`, `jss::ledger_hash` and `jss::close_time_iso` are grouped under one conditional check `if (ledger is validated) then {<fill these fields>}`

Reply:
thanks for a very good comment ! So I've spent some more time reading `AccountTx.cpp` and it turns out, this function only ever returns validated ledger. Which is actually rather visible near the top of `populateJsonResponse` function, where we do `response[jss::validated] = true;` . This also means that the line I added `response[jss::validated] = false;` would be a regression. I have removed it now, and will move other lines accordingly.

Reply:
@mDuo13 you might want to update documentation of `account_tx` - this function only ever returns transactions from validated ledger. It's not a new API version 2 behaviour.

Reply:
https://github.com/XRPLF/xrpl-dev-portal/pull/2250